File: 374-guess-number-higher-or-lower/374-guess-number-higher-or-lower.py
# The guess API is already defined for you.
# @param num, your guess
# @return -1 if my number is lower, 1 if my number is higher, otherwise return 0
# def guess(num: int) -> int:

class Solution:
    def guessNumber(self, n: int) -> int:
        start = 0
        end = n
        
        while start<=end:
            result = (start+end)//2
            
            if guess(result) == 0:
                return result
            elif guess(result) == -1:
                end = result -1
            else:
                start = result + 1
        

File: 374-guess-number-higher-or-lower/374-guess-number-higher-or-lower.ts
/**
 * Forward declaration of guess API.
 * @param {number} num   your guess
 * @return 	     -1 if num is higher than the picked number
 *			      1 if num is lower than the picked number
 *               otherwise return 0
 * var guess = function(num) {}
 */


function guessNumber(n: number): number {
    let left = 1, right = n
    while(left <= right){
        const mid = Math.floor((left + right)/2)
        const result = guess(mid)
        if(result === 0) return mid
        else if(result === 1){
            left = mid + 1
        }else{
            right = mid - 1
        }
    }

    return 0
};